addons/hr/views/hr_department_views.xml

```diff
@@ -43,12 +44,15 @@
             <field name="name">hr.department.tree</field>
             <field name="model">hr.department</field>
             <field name="arch" type="xml">
-                <tree string="Companies" sample="1">
+                <tree string="Companies" sample="1" editable="bottom">
+                    <field name="company_id" column_invisible="True"/>
+                    <field name="name"/>
                     <field name="display_name"/>
-                    <field name="company_id" groups="base.group_multi_company"/>
+                    <field name="company_id" options="{'no_create': True}" groups="base.group_multi_company"/>
```

Comment:
Maybe also make it readonly? It seems very unlikely to me that you'd want to change the company of a department.